orchestration/dagster_orchestration/hca_orchestration/tests/test_pipelines.py

```diff
@@ -14,25 +18,47 @@ def config_path(relative_path):
 
 
 class PipelinesTestCase(unittest.TestCase):
-    def run_pipeline(self, pipeline, config_name, *execution_args, **execution_kwargs):
-        config_dict = load_yaml_from_globs(
-            config_path(config_name)
-        )
+    def run_pipeline(self, pipeline, config_dict, mode, *execution_args, **execution_kwargs):
         return execute_pipeline(
             pipeline,
             *execution_args,
-            mode='test',
+            mode=mode,
             run_config=config_dict,
             **execution_kwargs
         )
 
+    def run_pipeline_with_yaml_config(self, pipeline, config_name, mode='test', *execution_args, **execution_kwargs):
+        config_dict = load_yaml_from_globs(
+            config_path(config_name)
+        )
+        return self.run_pipeline(pipeline, config_dict, mode, *execution_args, **execution_kwargs)
+
+    @pytest.mark.e2e
+    def test_stage_data_local_e2e(self):
+        test_id = f'test-{uuid.uuid4()}'
+
+        config = load_yaml_from_globs(config_path('stage_data_local_e2e.yaml'))
+        config['solids']['clear_staging_dir']['config']['staging_prefix_name'] = f'local-stage-data/{test_id}'
+        config['solids']['pre_process_metadata']['config']['staging_prefix_name'] = f'local-stage-data/{test_id}'
```

Comment:
instead of manually altering the config, we can generate a [preset](https://docs.dagster.io/_apidocs/presets) from the base config and use the `with_additional_config` method to generate a version with this extra config by passing the new config as a dict. example below:
```python
        base_test_preset = PresetDefinition.from_files("test_stage_data_e2e", config_files=config_path('stage_data_local_e2e.yaml'), mode='local')
        test_preset = base_test_preset.with_additional_config({'solids': {'clear_staging_dir': {'config': {'staging_prefix_name': f'local-stage-data/{test_id}'}}, 'pre_process_metadata': {'config': {'staging_prefix_name': f'local-stage-data/{test_id}'}}}})
```

then you can run the pipeline with `self.run_pipeline(stage_data, preset="test_stage_data_e2e")`, and it'll pull in the config you specified, though you'll need to make some alterations to make `config_dict` and `mode` optional fields, since `mode` is mutually exclusive with `preset`.

Reply:
i haven't played around with this much, so it's possible that this may not be exactly the syntax/structure for this, but i do think using presets here is the right approach.

Reply:
alternatively, if we find that the "preset" construct is more of a hindrance than a help, dagster ships with a tool to merge configs into each other: `from dagster.utils.merger import deep_merge_dicts`

Reply:
We spoke about this offline but sharing here for anyone else following: the preset approach is a good idea but won't fit this use-case. Specifically, we need to be able to determine the test_id and use for downstream test assertions. The preset approach forces us to set the test_id at pipeline-definition-time which precludes using it downstream in assertions. So, I've opted for the dict deep merge approach.